package DBToCS;

enum MsgID
{
	eCheckNickNameLogicCallBack = 1;
	eQueryUser_DBCallBack = 2;
	eBuyGoodsCallBack = 3;
	eChangeNickname_CallBack = 4;
	eQueryNotice_CallBack = 5;
	eMail_CallBack = 6;
	eQueryAllAccount_CallBack = 7;
	eQueryCDKeyEvents_CallBack = 8;
	eQueryAllUser_CallBack = 9;
}

message CDKeyEvents
{
	optional MsgID mgsid = 1 [default = eQueryCDKeyEvents_CallBack];
	message event_info{
		optional int32 id = 1;
		optional string title = 2;
		optional int32 platform = 3;
		optional uint64 end_time = 4;
		optional string content = 5;
		optional int32 gift_type = 6;
		optional string item_info = 7;
		optional int32 code_num = 8;
		optional int32 code_len = 9;
	}
	repeated event_info info = 2;
	optional int32 nsid = 3;
}

message QueryAllAccount
{
	optional MsgID mgsid = 1 [default = eQueryAllAccount_CallBack];
	message Account
	{
		optional string user_name = 1;
		optional int32 csid = 2;
		optional int32 sdkid = 3;
		optional string nickname = 4;
		optional int64 guid = 5;
	}
	repeated Account account = 2;
}

message MailCallBack
{
	optional MsgID mgsid = 1 [default = eMail_CallBack];
	optional int32 mailid = 2;
	optional int32 mailtype = 3;
	optional int32 channel = 4;
	optional string title = 5;
	optional string content = 6;
	optional string gift = 7;
	optional string sender = 8;
	optional string createtime = 9;
	optional int64 objid = 10;
}

message QueryNotice
{
	optional MsgID mgsid = 1 [default = eQueryNotice_CallBack];
	message Notice
	{
		optional uint32 id = 1;
		optional string title = 2;
		optional int32 eflag = 3;
		optional int32 estate = 4;
		optional int32 priority = 5;
		optional string notice = 6;
		optional uint64 star_time = 7;
		optional uint64 end_time = 8;
		optional uint32 platform = 9;
	}
	repeated Notice notice_info = 2;
}

message ChangeNickname
{
	optional MsgID mgsid = 1 [default = eChangeNickname_CallBack];
	optional uint64 guid = 2;
	optional bool reflag = 3;
	optional string newnickname = 4;
	optional string oldnickname = 5;
}

message BuyGoods
{
	optional MsgID mgsid = 1 [default = eBuyGoodsCallBack];
	optional string buygoodsmag = 2;
}

message CheckNameRsp
{
	optional MsgID mgsid = 1 [default = eCheckNickNameLogicCallBack];
	optional string completeInfo = 2;
	optional int32 gsid = 3;
	optional int32 gcnetid = 4;
	optional int32 gsnetid = 5;
	optional int32 checkres = 6;
}

message HeroCfg
{
	optional uint32 commodityid = 1;
	optional int64 buytime = 2;
	optional int64 expiredtime = 3;
}

message RuneInfo
{
	optional string bagStr = 1;
	optional string slotStr = 2;
}

message RSinfo{
	optional uint64 related_id = 1;
	optional uint32 related_header = 2;
	optional string related_name = 3;
	optional uint32 relation = 4;
	optional uint32 related_vip = 5;
}

message ItemInfo
{
	optional int32 item_id = 1;
	optional uint32 item_num = 2;
	optional int32 buy_time = 3;
	optional int32 end_time = 4;
} 

message MailInfo
{
	optional int32 mailid = 1;
	optional int32 state = 2;
}
message QueryUser
{
	optional MsgID msgid = 1 [default = eQueryUser_DBCallBack];
	optional string db = 2;
	optional string login = 3;
	optional int32	gsid = 4;
	optional int32 gcnetid = 5;
	optional string nickname = 6;
	repeated HeroCfg herocfg = 7;
	repeated RuneInfo runeinfo = 8;
	repeated RSinfo rsinfo = 9;
	repeated ItemInfo item_info = 10; 
	repeated MailInfo mail_info = 11;
	optional string guidestr = 12;
	optional string taskData = 13;
}
message QueryAllUser
{
	optional MsgID msgid = 1 [default = eQueryAllUser_CallBack];
	repeated QueryUser sQueryUser = 2;
}
